lib/vnet/app_resolver.go

```diff
@@ -257,15 +258,15 @@ func (r *TCPAppResolver) resolveTCPHandlerForCluster(
 	if err != nil {
 		// Don't return an unexpected error so we can try to find the app in different clusters or forward the
 		// request upstream.
-		slog.InfoContext(ctx, "Failed to list application servers.", "error", err)
+		log.InfoContext(ctx, "Failed to list application servers.", "error", err)
 		return nil, ErrNoTCPHandler
 	}
 	if len(resp.Resources) == 0 {
 		// Didn't find any matching app, forward the request upstream.
 		return nil, ErrNoTCPHandler
 	}
 	app := resp.Resources[0].GetApp()
-	appHandler, err := r.newTCPAppHandler(ctx, profileName, leafClusterName, app)
+	appHandler, err := r.newTCPAppHandler(ctx, log, profileName, leafClusterName, app)
```

Comment:
Hmmm - I don't love passing `log` in here because `newTCPAppHandler` also has access to `r.log` and it gets confusing to try to understand where you should be passing a logger around and where you should be using the logger on the method receiver.

Reply:
Yeah, I know. But this `log` already has all the extra fields specified and they're useful both in the context of `resolveTCPHandlerForCluster`, as well as within a TCP app handler itself. I didn't want to repeat them twice, but I suppose doing so would at least reduce the confusion around which logger to use. So I guess that would be the way forward?